Library/SunCalc.swift

```diff
@@ -0,0 +1,94 @@
+import JavaScriptCore
+import Foundation
+
+enum SunPhase {
+    case nightStart
+    case dawn
+    case sunrise
+    case solarNoon
+    case sunset
+    case dusk
+    case nightEnd
+}
+
+struct SunCalc {
+    private(set) var date: Date
+    private(set) var timeZone: TimeZone
+
+    private(set) var dawn: Date
+    private(set) var dusk: Date
+    private(set) var goldenHour: Date
+    private(set) var goldenHourEnd: Date
+    private(set) var nadir: Date
+    private(set) var nauticalDawn: Date
+    private(set) var nauticalDusk: Date
+    private(set) var night: Date
+    private(set) var nightEnd: Date
+    private(set) var solarNoon: Date
+    private(set) var sunrise: Date
+    private(set) var sunriseEnd: Date
+    private(set) var sunset: Date
+    private(set) var sunsetStart: Date
+
+    init(date: Date, timeZone: TimeZone, latitude: Double, longitude: Double, bundle: Bundle = Bundle.main) {
+        self.date = date
+        self.timeZone = timeZone
+
+        let path = bundle.path(forResource: "suncalc", ofType: "js")!
+        let contents = try! String(contentsOfFile: path)
+        let context = JSContext()!
+        context.evaluateScript(contents)
+
+        let function = context.objectForKeyedSubscript("getTimes")!
```

Comment:
Aaah it's the `gtTimes` function in javaScript!

Maybe just calling it `getTimesWithJavaScriptFunction` ?